pkg/cmd/repo/fork/fork.go

```diff
@@ -317,8 +324,28 @@ func forkRun(opts *ForkOptions) error {
 			}
 		}
 		if cloneDesired {
-			forkedRepoURL := ghrepo.FormatRemoteURL(forkedRepo, protocol)
-			cloneDir, err := gitClient.Clone(ctx, forkedRepoURL, opts.GitArgs)
+			// Allow injecting alternative BackOff in tests.
+			if opts.BackOff == nil {
+				bo := backoff.NewExponentialBackOff()
+				bo.Multiplier = 1.1
+				bo.MaxInterval = 10 * time.Second
+				bo.MaxElapsedTime = 5 * time.Minute
+				opts.BackOff = bo
+			}
+
+			cloneDir, err := backoff.RetryWithData(func() (string, error) {
+				forkedRepoURL := ghrepo.FormatRemoteURL(forkedRepo, protocol)
+				dir, err := gitClient.Clone(ctx, forkedRepoURL, opts.GitArgs)
+				if err == nil {
+					return dir, err
+				}
+				var execError errWithExitCode
+				if errors.As(err, &execError) && execError.ExitCode() == 128 {
+					return "", err
+				}
+				return "", backoff.Permanent(err)
+			}, backoff.WithContext(backoff.WithMaxRetries(opts.BackOff, 3), ctx))
```

Comment:
Will users see an indication of the failures/retries?

Since I trigger this problem regularly, I'd love to have a sense of how many retries I'm spending.

Reply:
As of right now, no they will not. Could be added in though.

Reply:
I think it's worth adding. That'd help inform decisions about the right time values.

Alternatively, I suppose I could try using this version of `gh` w/ that logging and see what they say...

Lemme see if I can manage to make that work locally...